variantlib/resolver/sorting.py

```diff
@@ -0,0 +1,218 @@
+from __future__ import annotations
+
+import logging
+import sys
+
+from variantlib.errors import ValidationError
+from variantlib.models.validators import validate_instance_of
+from variantlib.models.validators import validate_list_of
+from variantlib.models.variant import VariantDescription
+from variantlib.models.variant import VariantFeature
+from variantlib.models.variant import VariantProperty
+
+logger = logging.getLogger(__name__)
+
+
+def get_property_priority(
+    vprop: VariantProperty,
+    property_priorities: list[VariantProperty] | None,
+) -> int:
+    """
+    Get the property priority of a `VariantProperty` object.
+
+    :param vprop: `VariantProperty` object.
+    :param property_priorities: ordered list of `VariantProperty` objects.
+    :return: Property priority of the `VariantProperty` object.
+    """
+    validate_instance_of(vprop, VariantProperty)
+
+    if property_priorities is None:
+        return sys.maxsize
+    validate_instance_of(property_priorities, list)
+    validate_list_of(property_priorities, VariantProperty)
+
+    _property_priorities = [vprop.property_hash for vprop in property_priorities]
+
+    # if not present push at the end
+    try:
+        return _property_priorities.index(vprop.property_hash)
+    except ValueError:
+        return sys.maxsize
```

Comment:
Interesting idea. Not sure how clean it is, but I don't see any immediate issues.